pkg/controllers/logging.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"runtime/debug"
+
+	"github.com/go-logr/logr"
+	"github.com/sirupsen/logrus"
+)
+
+// LogrusSink is an adapter to allow the use of logrus with logr, as required by k8s controller-runtime.
+type LogrusSink struct {
+	logger        *logrus.Logger
+	names         []string
+	keysAndValues []interface{}
+}
+
+func NewLogrusSink(logger *logrus.Logger) *LogrusSink {
+	return &LogrusSink{
+		logger: logger,
+	}
+}
+
+// Init receives optional information about the logr library for LogSink
+// implementations that need it.
+func (sink *LogrusSink) Init(info logr.RuntimeInfo) {
+}
+
+// Enabled tests whether this LogSink is enabled at the specified V-level.
+// For example, commandline flags might be used to set the logging
+// verbosity and disable some info logs.
+func (sink *LogrusSink) Enabled(level int) bool {
+	// controller-runtime uses V=0 for Info and V=1 for Debug: https://github.com/kubernetes-sigs/controller-runtime/blob/master/TMP-LOGGING.md
+	switch level {
+	case 0:
+		return sink.logger.IsLevelEnabled(logrus.InfoLevel)
+	default:
+		return sink.logger.IsLevelEnabled(logrus.DebugLevel)
+	}
+}
+
+// Info logs a non-error message with the given key/value pairs as context.
+// The level argument is provided for optional logging.  This method will
+// only be called when Enabled(level) is true. See Logger.Info for more
+// details.
+func (sink *LogrusSink) Info(level int, msg string, keysAndValues ...interface{}) {
+	entry := sink.logger.WithField("name", sink.names)
+
+	if keysAndValues != nil {
+		entry.WithField("values", keysAndValues)
+	}
+
+	entry.Info(msg, keysAndValues)
+}
+
+// Error logs an error, with the given message and key/value pairs as
+// context.  See Logger.Error for more details.
+func (sink *LogrusSink) Error(err error, msg string, keysAndValues ...interface{}) {
+	entry := sink.logger.WithField("name", sink.names).WithError(err)
```

Comment:
will we possible to add name in front of error msg?
e.g. 
`time="2023-02-09T21:01:41Z" level=error msg="unable to get kubeconfig[]" error="invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable" name="[controller-runtime client config]"`
->
`time="2023-02-09T21:01:41Z" level=error msg="[controller-runtime client config] unable to get kubeconfig[]" error="invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable"`
(add name directly in front of the msg)
or
`time="2023-02-09T21:01:41Z" level=error name="[controller-runtime client config] msg="unable to get kubeconfig[]" error="invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable"`

It might be easier to distinguish the controller and verification log.

Reply:
The logrus format isn't quite this flexible with the default formatter (https://github.dev/sirupsen/logrus/blob/master/text_formatter.go#L132)

I've inlined the name into the beginning of the `msg` (if the name exists) as requested and used `.` as a separator to match the previous output a bit more:

```
time="2023-02-13T18:40:54Z" level=error msg="[controller-runtime.client.config] unable to get kubeconfig" error="invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable" name=controller-runtime.client.config
```

Note: the `name` property still exists for filtering logs. There are tests added now, so this should be easier to understand & change in the future if we need to